File: Data Structures & Algorithms/string-encode-and-decode/submission-0.py
class Solution:

    def encode(self, strs: List[str]) -> str:
        result = ""

        for s in strs:
            result += s
            result += ';'
        
        return result

    def decode(self, s: str) -> List[str]:
        l = []
        start = 0

        for i, c in enumerate(s):
            if c == ';':
               l.append(s[start:i])
               start = i + 1


        return l

File: Data Structures & Algorithms/anagram-groups/submission-0.cpp
class Solution {
   public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        unordered_map<string, vector<string>> s = {};

        for (auto str: strs) {
            string sorted_str = str;

            sort(sorted_str.begin(), sorted_str.end());
            s[sorted_str].push_back(str);
        }

        vector<vector<string>> result = {};
        for (auto& p : s) {
            result.push_back(p.second);
        }

        return result;
    }
};
